pkg/server/render.go

```diff
@@ -60,9 +61,26 @@ func (ctrl *Controller) renderHandler(w http.ResponseWriter, r *http.Request) {
 		out = &storage.GetOutput{Tree: tree.New()}
 	}
 
-	fs := out.Tree.FlamebearerStruct(p.maxNodes)
-	res := renderResponse(fs, out)
-	ctrl.writeResponseJSON(w, res)
+	switch p.format {
+	case "json":
+		fs := out.Tree.FlamebearerStruct(p.maxNodes)
+		res := renderResponse(fs, out)
+		ctrl.writeResponseJSON(w, res)
+	case "pprof":
+		pprof := out.Tree.PprofStruct(&tree.PprofMetadata{
+			Type: out.SpyName, // TODO
+			Unit: out.Units,
+		})
+		out, err := proto.Marshal(pprof.Pprof())
+		if err == nil {
+			ctrl.writeResponseFile(w, "profile.pprof", out)
+		} else {
+			ctrl.writeInternalServerError(w, err, "")
+		}
+	case "collapsed":
+		collapsed := out.Tree.Collapsed()
+		ctrl.writeResponseFile(w, "profile.collapsed", []byte(collapsed))
```

Comment:
I would maybe call it `profile.collapsed.txt` so that it's clear this is plaintext. And I guess the same comment about name from above applies here.